eZ/Publish/Core/Persistence/Legacy/Content/Location/Gateway/DoctrineDatabase.php

```diff
@@ -631,6 +657,30 @@ function ($pathString) use ($query, $handler) {
         $query->prepare()->execute();
     }
 
+    private function buildHiddenSubtreeQuery(string $selectExpr): QueryBuilder
+    {
+        $query = $this->connection->createQueryBuilder();
+        $expr = $query->expr();
+        $query
+            ->select($selectExpr)
+            ->from('ezcontentobject_tree', 't')
+            ->leftJoin('t', 'ezcontentobject', 'c', 't.contentobject_id = c.id')
```

Comment:
why left join, not inner join?

Reply:
This goes for this comment and most others really:
I have cut&pasted code from ezplatform-kernel. I was thinking it would be beneficial for later changes if the code in ezplatform-kernel and ezpublish-kernel are more or less the same.. If you see no value in that, I can do all the changes here and below that you suggest..

So yes, this could be a inner join, but is cut&paste from [platform-kernel](https://github.com/ezsystems/ezplatform-kernel/blob/v1.3.14/eZ/Publish/Core/Persistence/Legacy/Content/Location/Gateway/DoctrineDatabase.php#L621)
In the kernel, this function is called several places, but I think it still could be a innerJoin there as well

Reply:
Interesting comment. ez4 and ez5 already have quite different ideas about what "zombie" (corrupted) objects are visible/editable, due to the sql queries in use being similar but not the same. And, make no mistake, in any eZ installation which has been running for many years, there are bound to be many zombies walking around (sadly this is very hard to tackle purely in php, and the best fix would be to actually introduce FKs in the DB, but I digress...). It would be good if there was a clear guideline and consistency in usage of left-join vs inner-join for all queries dealing with the content tables (contents, attributes, versions, classes, class attributes, nodes, names, url_aliases, custom attributes data)

Reply:
The guideline here is quite simple. We do not take into the account corrupted data, unless the feature itself is about them. Data corruption should be fixed elsewhere due to performance. Location always has a Content item, therefore left join is not the optimal solution. Moreover in this case we have constraints against Content items table (`ezcontentobject`) so even if somehow we have corrupted data, they won't be matched. Therefore the only thing this left join does is adding unnecessary overhead.

Reply:
Agree with the above. i just wanted to make sure the same thinking is applied consistently

Reply:
Changed to `join` in https://github.com/ezsystems/ezpublish-kernel/pull/3142/commits/66ee6a667394b4bb962d330baef104e4e07ef042